src/orders/components/OrderHistory/OrderHistory.tsx

```diff
@@ -1,25 +1,84 @@
 // @ts-strict-ignore
 import { FetchResult } from "@apollo/client";
 import { DashboardCard } from "@dashboard/components/Card";
+import { CopyableText } from "@dashboard/components/CopyableText/CopyableText";
 import Form from "@dashboard/components/Form";
-import {
-  Timeline,
-  TimelineAddNote,
-  TimelineEvent,
-  TimelineEventProps,
-  TimelineNote,
-} from "@dashboard/components/Timeline";
-import { OrderEventFragment, OrderNoteUpdateMutation } from "@dashboard/graphql";
+import { Pill } from "@dashboard/components/Pill";
+import { Timeline, TimelineAddNote } from "@dashboard/components/Timeline/Timeline";
+import { TimelineEvent } from "@dashboard/components/Timeline/TimelineEvent";
+import { TimelineNote } from "@dashboard/components/Timeline/TimelineNote";
+import { toActor } from "@dashboard/components/Timeline/utils";
+import { OrderEventFragment, OrderEventsEnum, OrderNoteUpdateMutation } from "@dashboard/graphql";
 import { SubmitPromise } from "@dashboard/hooks/useForm";
 import { ORDER_EVENTS_DOCS_URL } from "@dashboard/links";
-import { Box, vars } from "@saleor/macaw-ui-next";
+import { orderUrl } from "@dashboard/orders/urls";
+import { Box, Text, vars } from "@saleor/macaw-ui-next";
 import { FormattedMessage, useIntl } from "react-intl";
+import { Link } from "react-router-dom";
 
-import ExtendedTimelineEvent from "./ExtendedTimelineEvent";
+import { ExtendedTimelineEvent } from "./ExtendedTimelineEvent";
 import { HistoryComponentLoader } from "./HistoryComponentLoader";
-import LinkedTimelineEvent from "./LinkedTimelineEvent";
 import { getEventMessage } from "./messages";
-import { getEventSecondaryTitle, isTimelineEventOfType } from "./utils";
+import { OrderHistoryDate } from "./OrderHistoryDate";
+import { groupEventsByDate, isTimelineEventOfType } from "./utils";
+
+// Date group header component with internationalized labels
+const DateGroupHeader = ({ groupKey }: { groupKey: string }) => {
+  const intl = useIntl();
+
+  const getLabel = (key: string): string => {
+    switch (key) {
+      case "TODAY":
+        return intl.formatMessage({
+          id: "zWgbGg",
+          defaultMessage: "Today",
```

Comment:
suggestion: missing description

```suggestion
          defaultMessage: "Today",
          description: "date group header",
```